src/test/ui/type/type-path-err-node-types-2.stderr

```diff
@@ -0,0 +1,14 @@
+error[E0282]: type annotations needed for the closure `fn(_, _) -> i32`
+  --> $DIR/type-path-err-node-types-2.rs:7:13
+   |
+LL |     let _ = |a, b: _| -> _ { 0 };
+   |             ^^^^^^^^^^^^^^^^^^^^ cannot infer type for closure `[closure@$DIR/type-path-err-node-types-2.rs:7:13: 7:33]`
+   |
+help: give this closure an explicit return type without `_` placeholders
+   |
+LL |     let _ = |a, b: _| -> i32 { 0 };
+   |                          ^^^
```

Comment:
This is a regression. I need to see if we can avoid it by finding a way to do without the `delay_span_bug` for `?Sized` locals hack.